core/src/main/java/org/apache/calcite/adapter/enumerable/RexToLixTranslator.java

```diff
@@ -838,6 +848,13 @@ public static Expression translateLiteral(
     return Expressions.constant(value2, javaClass);
   }
 
+  private static MethodCallExpression adjustDecimalScale(int scale,
+      Expression expression) {
+    assert expression.getType() == BigDecimal.class;
+    return Expressions.call(expression, BuiltInMethod.BIG_DECIMAL_SET_SCALE.method,
+        Expressions.constant(scale), Expressions.constant(RoundingMode.HALF_UP));
```

Comment:
Why did we pick the `RoundingMode.HALF_UP`? Is it something mandated by the SQL standard? Is it something commonly used by other DBMS? Do we need to document this behavior in the website?

Reply:
I've been thinking about this point for quite a while, `RoundingMode.HALF_UP` might not be the best, maybe `RoundingMode.HALF_EVEN` would be better, but I'm not really sure, so I refer to the
`org.apache.calcite.util.NumberUtil#rescaleBigDecimal` to keep the behavior consistent with the original. If we plan to tweak the use of `RoundingMode`, I think we'll probably need to look at all the places where it is already being used and refer to more DBMSs to make the tradeoffs.

Reply:
As long as we are consistent with other places in Calcite that's fine with me.